internal/v2/s2av2.go

```diff
@@ -29,11 +29,12 @@ type s2av2TransportCreds struct {
 	serverName string
 	s2av2Address string
 	tokenManager *tokenmanager.AccessTokenManager
+	verificationMode s2av2pb.ValidatePeerCertificateChainReq_VerificationMode
 }
 
 // NewClientCreds returns a client-side transport credentials object that uses
 // the S2Av2 to establish a secure connection with a server.
-func NewClientCreds(s2av2Address string) (credentials.TransportCredentials, error) {
+func NewClientCreds(s2av2Address string, verificationMode s2av2pb.ValidatePeerCertificateChainReq_VerificationMode) (credentials.TransportCredentials, error) {
```

Comment:
We don't want to expose the proto directly to the users, so I think the right way to do this is:

1. Add a "verification mode" enum to the options file: https://github.com/google/s2a-go/blob/main/s2a_options.go

2. Add a "verification mode" field to each of the client/server options, which defaults to the "connecting to Google" value

3. Plumb this value through to cert_verifier.go, and then convert to the proto enum when preparing the SessionReq message.

Does that sound reasonable?

Reply:
Thanks for pointing this out! In this PR, I added a verificationMode field to the client and server options in s2a_options, but I did expose the proto, I'll change it to the enum.